src/layers/SchematicLayer.ts

```diff
@@ -305,7 +323,46 @@ export class SchematicLayer<T extends SchematicData> extends PixiLayer<T> {
     }
 
     const { exaggerationFactor } = this.options as SchematicLayerOptions<T>;
-    const { holeSizes, casings, cements, completion, symbols, pAndA } = this.data;
+    const { holeSizes, casings, cements, completion, symbols, pAndA, perforations } = this.data;
+
+    const shouldStartAtHoleDiameter = perforations.filter((perf) =>
+      foldPerforationSubKind(
+        {
+          Perforation: () => true,
+          OpenHole: () => false,
+          OpenHoleGravelPack: () => true,
+          OpenHoleFracPack: () => false,
+          OpenHoleScreen: () => true,
+          CasedHoleFracturation: () => false,
+          CasedHoleGravelPack: () => false,
+          CasedHoleFracPack: () => false,
+        },
+        perf.subKind,
+      ),
+    );
+
+    const shouldStartAtCasingDiameter = perforations.filter(
+      (perf) =>
+        !foldPerforationSubKind(
+          {
+            Perforation: () => true,
+            OpenHole: () => false,
+            OpenHoleGravelPack: () => true,
+            OpenHoleFracPack: () => false,
+            OpenHoleScreen: () => true,
+            CasedHoleFracturation: () => false,
+            CasedHoleGravelPack: () => false,
+            CasedHoleFracPack: () => false,
+          },
+          perf.subKind,
+        ),
+    );
```

Comment:
Should these be moved to schematic interface file instead? 🤔 
Think I saw similar folds there.